keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -162,10 +171,9 @@ class ArtifactListener(
     return ArtifactStatus.valueOf(status)
   }
 
-  private val artifactTypeNames by lazy { ArtifactType.values().map(ArtifactType::name) }
+  private fun Artifact.type() = ArtifactType.valueOf(type.toUpperCase())
```

Comment:
Why a `fun` rather than a `val`?

Reply:
`artifactTypeNames` just moved down to line 176. This is an additional function.

Reply:
Yeah, but `type` seems weird as a function to me, rather than a property

Reply:
Oh, the artifact in kork already has a property `type`, so this function is to parse the `ArtifactType` from that string value.